### PR TITLE
Extend README instructions

### PR DESCRIPTION
**Pull request by skisela**
Extend README information about what useful output rebase-helper produces.

### CODE REVIEW

**Comment by nforro** [comment]
This is good, but I think a table listing all the result files/directories would be better. Also it should be mentioned which files/directories are present always, and which can be missing due to failed rebase.

**Comment by nforro** [comment]
I don't think **Phase of generation** is a good title, something like **Notes** would be probably better.

File: README.md [hunk 1]
```diff
@@ -31,6 +31,20 @@ Alternatively, you can run **rebase-helper** in a container:
...
+## What do I get from **rebase-helper**?
+
+**rebase-helper** creates *rebase-helper-results* directory which contains the following useful items:
+
+| Name											| Description   | Phase of generation |
+| :-------------------------| :-------------| :-------------------|
+| *report.txt*							| complete **rebase-helper** summary output with all important summary information | always |
+| *changes.patch*						| patch file containing changes diff against original files, ready to be applied to the dist-git | always contains version changes and changelog entry, after successful patching contains also patch changes |
+| *checkers/*								| directory containing detailed output from each checker that was used during the run      | SOURCE checkers output after sources extraction, SRPM checkers output after successful SRPM builds, RPM checkers output after successful RPM builds  |
+| *rebased-sources/*				| modified dist-git files ready to be used (Primarily used for generating *changes.patch*) | after successful patching |
```
Comment (nforro) [inline README.md]: This is not true, some files will be there even if patching failed.

File: README.md [hunk 2]
```diff
@@ -31,6 +31,20 @@ Alternatively, you can run **rebase-helper** in a container:
...
+| *old-build/*							| directory containing complete old (S)RPM build with its logs | after successful old build |
```
Comment (nforro) [inline README.md]: Same here, build logs will be there even if build failed.

**Comment by nforro** [comment]
What about this?

## What do I get from it?

**rebase-helper** always creates *rebase-helper-results* directory containing the following items:

| Path                      | Description                                                       |
|:------------------------- |:----------------------------------------------------------------- |
| *report.txt*              | summary report with all important information                     |
| *changes.patch*           | diff against original files, directly applicable to dist-git repo |
| *rebased-sources/*        | git repository with all modified files                            |
| *checkers/*               | reports from individual checkers that were run                    |
| *old-build/*              | logs and results of old (original) version build                  |
| *new-build/*              | logs and results of new (rebased) version build                   |
| *rebase-helper-debug.log* | log containing the most verbose output of a rebase                |

I decided to drop **Notes** because almost everything is always present, or it's obvious why it's not.

File: README.md [hunk 3]
```diff
@@ -31,6 +31,20 @@ Alternatively, you can run **rebase-helper** in a container:
 
 See [docker reference](https://rebase-helper.readthedocs.io/en/latest/user_guide/rebasing_in_container.html) for more information.
 
+## What do I get from **rebase-helper**?
```
Comment (nforro) [inline README.md]: All other headings refer to `rebase-helper` as `it`, I don't think this one should be an exception. Alternatively you want to replace `it` with `rebase-helper` in all the headings.

File: README.md [hunk 4]
```diff
@@ -31,6 +31,20 @@ Alternatively, you can run **rebase-helper** in a container:
...
+
+**rebase-helper** creates *rebase-helper-results* directory which contains the following useful items:
```
Comment (nforro) [inline README.md]: `useful` sounds a bit over the top to me. And what's wrong with `always`?

Comment (skisela) [reply README.md]: I do not understand the question. What is wrong with what `always`?

Comment (nforro) [reply README.md]: I was referring to my proposal in comment https://github.com/rebase-helper/rebase-helper/pull/477#issuecomment-380441060.

**Pull request merged**